tsdat/io/storage.py

```diff
@@ -158,21 +174,37 @@ def __get_date_str(file: Path) -> str:
 
     def _open_data_files(self, *filepaths: Path) -> List[xr.Dataset]:
         dataset_list: List[xr.Dataset] = []
-        for filepath in filepaths:
-            data = self.handler.reader.read(filepath.as_posix())
+        for filepath in filepaths:  # type: ignore
+            data = self.handler.reader.read(filepath.as_posix())  # type: ignore
             if isinstance(data, dict):
                 data = xr.merge(data.values())  # type: ignore
             dataset_list.append(data)
         return dataset_list
 
-    def _get_dataset_filepath(self, dataset: xr.Dataset, datastream: str) -> Path:
-        datastream_dir = self.parameters.storage_root / "data" / datastream
+    def _get_dataset_filepath(self, dataset: xr.Dataset, by_date: bool) -> Path:
```

Comment:
`by_date` is an attribute on `self` so it doesn't need to be passed as an argument here.